File: seeed_arm_console/sdk/cpp/examples/planner_client.cpp
#include "rebot_sdk/client.hpp"

#include <cstdlib>
#include <iostream>

int main(int argc, char** argv) {
    const std::string address = argc > 1 ? argv[1] : "127.0.0.1:50053";
    auto channel = grpc::CreateChannel(address, grpc::InsecureChannelCredentials());
    rebot::sdk::ArmPlannerClient planner(std::move(channel));
    rebot::sdk::PoseTarget target;
    target.position_m = {0.25, 0.0, 0.30};

    rebot::sdk::IKResult ik;
    auto status = planner.solve_ik(target, &ik);
    if (!status.ok() || !ik.success) {
        std::cerr << "IK failed: " << (status.ok() ? ik.reason : status.error_message()) << '\n';
        return EXIT_FAILURE;
    }
    rebot::sdk::PoseTarget goal;
    goal.position_m = {0.20, 0.05, 0.32};
    rebot::sdk::TrajectoryPlanResult plan;
    status = planner.plan_trajectory(target, goal, &plan);
    if (!status.ok() || !plan.success || plan.points.size() < 2) {
        std::cerr << "trajectory failed: " << (status.ok() ? plan.reason : status.error_message()) << '\n';
        return EXIT_FAILURE;
    }
    std::cout << "cpp_planner_sdk=OK ik_joints=" << ik.joint_position_rad.size()
              << " trajectory_points=" << plan.points.size()
              << " solver=" << plan.metadata.solver << '\n';
    return EXIT_SUCCESS;
}

File: seeed_arm_console/sdk/rust/examples/planner_client.rs
use rebot_arm_sdk::{ArmPlannerClient, IKOptions, PoseTarget, TrajectoryOptions};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let endpoint = std::env::var("ARM_PLANNER_GRPC_URL")
        .unwrap_or_else(|_| "http://127.0.0.1:50053".to_owned());
    let mut planner = ArmPlannerClient::connect(endpoint).await?;
    let result = planner
        .solve_ik(
            PoseTarget::new([0.25, 0.0, 0.30]),
            IKOptions {
                ..Default::default()
            },
        )
        .await?;
    println!(
        "ik success={} joints={} reason={}",
        result.success,
        result.joint_position_rad.len(),
        result.reason
    );
    if result.success {
        let plan = planner
            .plan_trajectory(
                PoseTarget::new([0.25, 0.0, 0.30]),
                PoseTarget::new([0.20, 0.05, 0.32]),
                TrajectoryOptions::default(),
            )
            .await?;
        println!(
            "trajectory success={} points={} solver={}",
            plan.success,
            plan.points.len(),
            plan.metadata.solver
        );
    }
    Ok(())
}
